cmd/gateway-main.go

```diff
@@ -179,6 +180,36 @@ func parseGatewayEndpoint(arg string) (endPoint string, secure bool, err error)
 	}
 }
 
+// Validate gateway arguments.
+func validateGatewayArguments(serverAddr, endpointAddr string) error {
+	if err := CheckLocalServerAddr(serverAddr); err != nil {
```

Comment:
@harshavardhana yes, this addresses the problem my PR fixes. You can close my PR in favour.